omc/core/model/board.py

```diff
@@ -0,0 +1,545 @@
+
+        return self._piece_pxl_hex
+
+    @property
+    def current_players(self) -> list[int]:
+        """
+        Gets the list of current players from board.
+
+        :return: List of the current player numbers
+        :rtype: list[int]
+        """
+
+        return list(self._board.current_players.keys())
+
+    @property
+    def current_coords(self) -> ndarray[int]:
+        """
+        Gets the current coordinates of the piece
+
+        :return: Current player coordinate
+        :rtype: ndarray[int]
+        """
+        return self._current_coords
+
+    @property
+    def player_controller(self) -> int:
+        """
+        Gets the current player controller.
+
+        :return: Number associated with player currently controlling the piece.
+        :rtype: int
+        """
+        return self._player_controller
+
+    def list_moves(self) -> list[ndarray[int] | None]:
+        """
+        Gives a list of all available moves for the piece.
+
+        :return: List of moves currently able to be performed by
+            the piece.
+        :rtype: list[Optional[ndarray[int]]]
+        """
+
+        raise NotImplementedError
+
+    def move(self, coords: ndarray[int]) -> bool:
+        """
+        Performs an action on the piece using arguments.
+
+        :param coords: Coordinates for the space of the attempted move.
+        :type coords: ndarray[int]
+        :return: True on successful move, False otherwise.
+        :rtype: bool
+        """
+
+        # Check if move is valid
+        if coords not in self.list_moves():
+            print("Invalid Move.")
+            return False
+
+        self._current_coords = coords
+        return True
+
+    def capture(self, new_player: int) -> bool:
+        """
+        Attempt to capture this piece, removing it from the board and giving
+        possession of the piece to `new_player`.
+
+        :param new_player: Player number for the player capturing this piece
+        :type new_player: int
+        :return: True on successful capture, False otherwise.
+        :rtype: bool
+        """
+        if self.set_player_control(new_player):
+            self._current_coords = None
+            self._board.remove_piece(self)
+            return True
+        return False
+
+    def set_player_control(self, new_player: int) -> bool:
+        """
+        Changes the player in control of this piece.
+
+        :param new_player: Player number taking control of the piece.
+        :type new_player: int
+        :return: True on successful action, False otherwise.
+        :rtype: bool
+        """
+
+        if new_player < 0:
+            print("Invalid Player Number (Negative).")
+            return False
+
+        valid_players = self.current_players
+
+        if new_player not in valid_players:
+            print(f"Invalid Player Number. (Valid Numbers: {valid_players})")
+            return False
+
+        self._player_controller = new_player
+        return True
```

Comment:
Now that players are their own class, this should update the player instances when changing player control
- Remove piece from player_controller.pieces
- Change player controller of the piece
- Add  piece to new_player.pieces